querymsc.py

```diff
@@ -505,5 +507,124 @@ def query_rain_rate_one_day_1_50(self, session):
                                   History.lon,
                                   History.station_id)
                  )
- 
-        return query
\ No newline at end of file
+  
+        return query
+
+class Gumbel:
+    """This class is meant to do all the fitting
+    of Gumbel distributions to extract design values.
+    
+    Attributes:
+        return_period (int): return period in years
+        min_fit (int): minimum number of years to consider
+            fitting Gumbel.
+    """
+
+    def __init__(self, return_period, min_fit=10):
+
+        self.return_period = return_period
+        self.min_fit = min_fit
+
+    def fit(self, x):
+        """Function to get L-moments to estimate the parameters
+        of a gumbel distribution. This method uses the 
+        right-skewed Gumbel distribution. Method mirrors 
+        Hosking 1990.
+        -----------------------------------
+        Args:
+            x (pandas Series): Series containing the annual
+                grouped extreme values for a given 
+                station over a range of years.
+        Returns:
+            (xi, alpha) (tuple): estimated parameters of gumbel
+                distribution if N_min criteria met
+            NaN (numpy NaN object): if N_min criteria is not met 
+        """
+
+        N = x.shape[0]
+
+        # euler-mascheroni constant
+        euler = 0.5772156649
+
+        if N >= self.min_fit:
+            # create fitted gumbel object to
+            # distribution of extreme values
+            paras = distr.gum.lmom_fit(x)
+
+            # extract L-moments
+            lmoments = distr.gum.lmom(nmom=2, **paras)
+
+            # calculate estimators of gumbel
+            # parameters
+            alpha = lmoments[1]/np.log(2)
+            xi = lmoments[0] - euler*alpha 
+
+            return xi, alpha
+        else:
+            return np.nan, np.nan
+
+    def get_design_value(self, xi, alpha):
+        """Get the design value from explicit expression
+        derived from a right-skewed Gumbel distribution.
+        See methods.pdf for more details. Separate from
+        Gumbel fitting in case user defines custom
+        estimators. 
+        -----------------------------------
+        Args:
+            xi (float): Estimated value of the location parameter
+                (first l-moment)
+            alpha (float): Estimated value of the scale parameter
+                (second l-moment).
+        Returns:
+            design_val (float): design value using Gumbel parameters
+        """
+
+        # return frequency, i.e. inverse of return period 
+        f_r = 1.0/self.return_period
+
+        # simplify long expression
+        simp = (1.0-f_r) + np.exp(-np.exp((xi/alpha)))
```

Comment:
Likewise, what if a user passes in an alpha of 0? And what happens in the next step if `simp` comes out as a negative number?